urllib3/poolmanager.py

```diff
@@ -335,6 +336,12 @@ def urlopen(self, method, url, redirect=True, **kw):
         if not isinstance(retries, Retry):
             retries = Retry.from_int(retries, redirect=redirect)
 
+        # Strip headers marked as unsafe to forward to the redirected location.
+        if (retries.remove_headers_on_redirect
```

Comment:
Optional nit: Would be equivalent to have

```python
        if not conn.is_same_host(redirect_location)):
            for header in retries.remove_headers_on_redirect:
                kw['headers'].pop(header, None)
```

Reply:
Only reason I chose to do this is because of the `TODO: socket.gethostname()` within `ConnectionPool.is_same_host()` in order to short-circuit a potential future network call if not needed. :)

Reply:
Good point. In that case, a comment would be wise so a future contributor doesn't simplify it. :)